RFS/src/main/java/com/rfs/common/GlobalMetadata.java

```diff
@@ -0,0 +1,67 @@
+package com.rfs.common;
+
+import org.apache.lucene.codecs.CodecUtil;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.nio.file.Path;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.dataformat.smile.SmileFactory;
+
+
+public class GlobalMetadata {
```

Comment:
This factory pattern seemed a little confusing to me initially. My anticipation was that this class would be called `GlobalMetadataFactory` and would have a static method which could determine the proper implementation to return, say `GlobalMetadataES710`, with this implementation having implemented say a `GlobalMetadata` interface that all implementation implement. I'm curious your thoughts, but I'm also fine to tackle this later on instead of now.

Reply:
I'm fine to agree to disagree here as I can logically see the intent and this will likely develop a lot in the near future.